### PR TITLE
Split tracing calls for TxProcessor and SafeService

### PR DESCRIPTION
**Pull request by Uxio0**
- Currently if tracing was enabled every RPC call from this service was performed to the tracing node, even if not required
- With this update, only tracing specific calls will be performed to the tracing node

### CODE REVIEW

File: safe_transaction_service/history/indexers/tx_processor.py [hunk 1]
```diff
@@ -82,9 +83,14 @@ class SafeTxProcessor(TxProcessor):
     Processor for txs on Safe Contracts v0.0.1 - v1.0.0
     """
 
-    def __init__(self, ethereum_client: EthereumClient):
+    def __init__(
+        self,
+        ethereum_client: EthereumClient,
+        ethereum_tracing_client: Optional[EthereumClient],
```
Comment (rmeissner) [inline safe_transaction_service/history/indexers/tx_processor.py:89]: Some docs documenting what happens if it is not set would be nice

File: safe_transaction_service/history/services/safe_service.py [hunk 2]
```diff
@@ -64,9 +64,13 @@ def del_singleton(cls):
 
 
 class SafeService:
-    def __init__(self, ethereum_client: EthereumClient, tracing_enabled: bool):
+    def __init__(
+        self,
+        ethereum_client: EthereumClient,
+        ethereum_tracing_client: Optional[EthereumClient],
```
Comment (rmeissner) [inline safe_transaction_service/history/services/safe_service.py:70]: Same as above

File: safe_transaction_service/history/services/safe_service.py [hunk 3]
```diff
@@ -220,10 +224,10 @@ def _get_next_internal_tx(self, internal_tx: InternalTx) -> Optional[InternalTx]
     def _get_parent_internal_tx(self, internal_tx: InternalTx) -> InternalTx:
         if parent_trace := internal_tx.get_parent():
             return parent_trace
-        if not self.tracing_enabled:
```
Comment (rmeissner) [inline safe_transaction_service/history/services/safe_service.py:223]: Was this field deprecated?

Comment (Uxio0) [reply safe_transaction_service/history/services/safe_service.py:223]: It's not required anymore, checking `if not self.ethereum_tracing_client` is enough

File: safe_transaction_service/history/indexers/tx_processor.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/history/indexers/tx_processor.py:89]: Updated in a the new commit

**Pull request merged**